src/expr/src/relation/func.rs

```diff
@@ -513,6 +513,95 @@ impl AggregateFunc {
+    /// values produced by `self`, and if necessary, a function to convert the resulting type.
+    /// If there does not exist a function that can combine multiple aggregates
+    /// of this type, this method returns `None`.
+    ///
+    /// Many aggregate functions can be combined, but doing so may require the use of
+    /// a different function, and potentially some type conversion to return to the intended
+    /// type. For example, `Count` aggregates can be combined with `SumInt64` to produce
+    /// a combined count, but they must also be converted back from its decimal output to
+    /// `Int64` to have the correct output type for `Count`.
+    pub fn outer_agg(&self) -> Option<(AggregateFunc, Option<crate::UnaryFunc>)> {
+        match self {
+            AggregateFunc::Count => Some((
+                AggregateFunc::SumInt64,
+                Some(crate::UnaryFunc::CastDecimalToInt64(0)),
+            )),
+            AggregateFunc::SumInt32 => Some((
+                AggregateFunc::SumInt64,
+                Some(crate::UnaryFunc::CastDecimalToInt64(0)),
+            )),
+            AggregateFunc::SumInt64 => Some((AggregateFunc::SumDecimal, None)),
+            // we don't yet have an aggregation to flat map jsonb lists into bigger lists
+            AggregateFunc::JsonbAgg => None,
+            _ => Some((self.clone(), None)),
+        }
+    }
+
+    /// Whether an aggregate always returns the same answer when evaluated on a
+    /// collection as when evaluated on the same collection but with duplicate
+    /// values removed.
+    ///
+    /// Suppose we have a collection `A=[1, 1, 2, 4, 4]`. The maximum value in
+    /// the collection does not change if we remove duplicate values. However,
+    /// the sum of the values in the collection will.
+    pub fn ignores_duplicates(&self) -> bool {
+        !matches!(
+            self,
+            AggregateFunc::MaxInt32
+                | AggregateFunc::MaxInt64
+                | AggregateFunc::MaxFloat32
+                | AggregateFunc::MaxFloat64
+                | AggregateFunc::MaxDecimal
+                | AggregateFunc::MaxBool
+                | AggregateFunc::MaxString
+                | AggregateFunc::MaxDate
+                | AggregateFunc::MaxTimestamp
+                | AggregateFunc::MaxTimestampTz
+                | AggregateFunc::MinInt32
+                | AggregateFunc::MinInt64
+                | AggregateFunc::MinFloat32
+                | AggregateFunc::MinFloat64
+                | AggregateFunc::MinDecimal
+                | AggregateFunc::MinBool
+                | AggregateFunc::MinString
+                | AggregateFunc::MinDate
+                | AggregateFunc::MinTimestamp
+                | AggregateFunc::MinTimestampTz
+                | AggregateFunc::Dummy
+        )
+    }
+
+    /// Whether the result applying the aggregate to a literal is
+    /// dependent on the specific number of rows in the input.
```

Comment:
Having read the comments and body of the method, I'm not sure what this does. Perhaps let's talk out what it does and we can capture the result in the comments.